src/Delegate/NotFoundDelegate.php

```diff
@@ -93,9 +102,14 @@ private function generatePlainTextResponse(ServerRequestInterface $request)
      */
     private function generateTemplatedResponse(ServerRequestInterface $request)
     {
+        $templateData = [
+            'request'  => $request,
+            'layout'  => $this->layout,
+        ];
+
         $response = $this->responsePrototype->withStatus(StatusCodeInterface::STATUS_NOT_FOUND);
         $response->getBody()->write(
-            $this->renderer->render($this->template, ['request' => $request])
+            $this->renderer->render($this->template, $templateData)
```

Comment:
Why add a new variable?

```php
$this->renderer->render($this->template, [
    'request'  => $request,
    'layout'  => $this->layout,
]);
```

Reply:
I will rework this point to make the parameter optional - or do you have a better idea? Another option would be that in the factory other optional parameters can be passed to the renderer no matter what...